samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/core/config/CoreConfig.java

```diff
@@ -0,0 +1,27 @@
+package io.eventdriven.ecommerce.core.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.eventdriven.ecommerce.core.events.InMemoryEventBus;
+import io.eventdriven.ecommerce.core.events.EventBus;
+import io.eventdriven.ecommerce.core.scopes.ServiceScope;
+import io.eventdriven.ecommerce.core.serialization.EventSerializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class CoreConfig {
+  @Bean
```

Comment:
FYI. Here you are using Spring's dependency container to achieve DI. A really good alternative in Java is [Dagger](https://dagger.dev/). The main advantage of Dagger is that it will resolve dependencies at compile time so you avoid surprises when running your code in production.

Reply:
Thanks, I saw this some time ago and forgot about that 👍  Do you know how popular it's in the community? I tried to minimise the number of dependencies and use the "agnostic" way. I created a dedicated issue for that https://github.com/oskardudycz/EventSourcing.JVM/issues/3.